server.go

```diff
@@ -1944,6 +1961,38 @@ func SendHeader(ctx context.Context, md metadata.MD) error {
 	return nil
 }
 
+// SetSendCompressor sets a compressor for outbound messages.
+// It must not be called after any event that causes headers to be sent
+// (see SetHeader for a complete list). Provided compressor is used when below
+// conditions are met:
+//
+//   - compressor is registered via encoding.RegisterCompressor
+//   - compressor name exists in the client advertised compressor names sent in
+//     :grpc-accept-encoding header.
+//
+// The context provided must be the context passed to the server's handler.
+//
+// The error returned is compatible with the status package.  However, the
+// status code will often not match the RPC status as seen by the client
+// application, and therefore, should not be relied upon for this purpose.
+//
```

Comment:
Given that the `sendCompress` field in the `Stream` (which is set from here using a call to `stream.SetSendCompress()`) is not protected by a mutex, should we add a note here saying this method is not safe to be called concurrently? [here](https://github.com/grpc/grpc-go/blob/638141fbb9c9f6a40b23fe073a582aaaddb43b4f/stream.go#L1467) is an example.

Reply:
Added ```It is not safe to call SetSendCompressor concurrently with SendHeader and SendMsg.```